### ISSUE TITLE
Segmentation fault on whoami_s after unbind

### ISSUE THREAD

**Issue by florentx** [pyldap/pyldap#29]
Hello,

I've hit this segfault in upstream python-ldap, in an interactive session.
Then I've found your project on Github, and it segfaulted again.

Of course it should give an error in this case, but not segfault.

``` python

>>> import ldap
>>> ldap.__version__
'2.4.21'
>>> conn = ldap.initialize('ldap://')
>>> 
>>> conn.unbind()
>>> conn.whoami_s()
Segmentation fault (core dumped)
```

Tested with pyldap 2.4.21 and upstream version 2.4.22 and 2.4.25.

The crash with upstream is slightly different:

``` python
>>> import ldap
>>> 
>>> print(ldap.__version__)
2.4.25
>>> conn = ldap.initialize('ldap://')
>>> 
>>> conn.unbind()
>>> conn.whoami_s()
python: whoami.c:68: ldap_whoami: Assertion '( (ld)->ldc->ldc_options.ldo_valid == 0x2 )' failed.
Aborted (core dumped)
```

(I've lost my SF password to report the issue upstream ...)

**Comment by florentx** [comment]
I just noticed it was reported on the mailing list on December 2nd 2015:
https://mail.python.org/pipermail/python-ldap/2015q4/003649.html

**Comment by tiran** [comment]
`l_ldap_whoami_s()` is missing the check `if (not_valid(self)) return NULL;`

**Comment by tiran** [comment]
Michael Ströder said: 'fix will be released in 2.4.26'

**Comment by florentx** [comment]
thank you for the bugfix and for reporting the issue upstream :+1:
